src/mscorlib/src/System/Text/StringBuilder.cs

```diff
@@ -347,9 +350,9 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
                             int chunkLength = chunk.m_ChunkLength;
     
                             // Check that we will not overrun our boundaries. 
-                            if ((uint)(chunkLength + chunkOffset) <= ret.Length && (uint)chunkLength <= (uint)sourceArray.Length)
+                            if ((uint)(chunkLength + chunkOffset) <= (uint)ret.Length && (uint)chunkLength <= (uint)sourceArray.Length)
```

Comment:
Nice catch. I noticed that the rhs uint cast was missing before but did not bother submitting a PR for it.